cmd/askd/handlers/form_submission.go

```diff
@@ -197,55 +192,59 @@ func (formSubmissionHandle) Search(c *app.Context) error {
 		return err
 	}
 
-	if csv := c.Request.URL.Query().Get("csv"); csv != "" {
-		results.CSVURL = fmt.Sprintf("http://%v%v/csv", c.Request.Host, c.Request.URL.Path)
-	}
-
 	c.Respond(results, http.StatusOK)
 
 	return nil
 }
 
-// SearchCSV retrieves a set of FormSubmission's based on the search params
-// provided in the query string. It also generates a CSV with the replies.
+// Download retrieves a set of FormSubmission's based on the search params
+// provided in the query string and generates a CSV with the replies.
 // 200 Success, 400 Bad Request, 404 Not Found, 500 Internal
-func (formSubmissionHandle) SearchCSV(c *app.Context) error {
-	formID := c.Params["form_id"]
+func (formSubmissionHandle) Download(c *app.Context) error {
 
-	limit := 0
-	skip := 0
+	if c.Request.URL.Query().Get("download") == "true" { // Generates and returns the CSV file.
 
-	opts := submission.SearchOpts{
-		Query:    c.Request.URL.Query().Get("search"),
-		FilterBy: c.Request.URL.Query().Get("filterby"),
-	}
+		formID := c.Params["form_id"]
```

Comment:
Here it shouldn't, as it will only get to that handler when the path param exists.